### PR TITLE
Terra firma overshoot pi

### PR DESCRIPTION
**Pull request by ledm**
Added a new tool to only plot a limited time range (because Colin asked for it). 

Meanwhile, added some ymls to plot recent Terrafirma runs.

### CODE REVIEW

**Comment by valeriupredoi** [comment]
now `temperature_debug.yml` is needed :bug:

**Comment by valeriupredoi** [comment]
@ledm I don't need that file (`temperature_debug.yml`) but the test does, it could be the test is wrong and it needs fixing - is that a file that is needed for the run? Likewise `temperature.yml` :beer:

**Comment by ledm** [comment]
So, I changed the key_list/debug.yml, but you were using that to perform tests? So we've got a file that has multiple purposes? 

Perhaps the solution is to copy the debug.yml into a new file called github_test.yml or something and use that?

**Comment by ledm** [comment]
I've also added a script to run this on lotus and some documentation to the README.md.

**Comment by valeriupredoi** [comment]
sorry @ledm buds, lemme fix the test here then I'll review - been busy with fleeing from France last week :laughing:

**Comment by ledm** [comment]
Absolutely no worries! No rush mate.

**Comment by valeriupredoi** [comment]
fixed the tests - and the good news is that we now have two working (very basic) end-to-end analyses running...end to end :grin: timeseries and compare, they actually run on mock test data and finish yayy

File: README.md [hunk 1]
```diff
@@ -445,6 +448,51 @@ This  gnerated an HTML5 mobile-friendlyt report, summarising the output of a
 single model run.
 
 
+
+Running a job on JASMIN's batch system, LOTUS
+=============================================
+
+Instead of running these jobs interactively on the command line, 
+it is possible to submit them as batch jobs to jasmin's LOTUS machine.
```
Comment (valeriupredoi) [inline README.md:456]:
```suggestion
it is possible to submit them as batch jobs to JASMIN's LOTUS machine.
```

File: README.md [hunk 2]
```diff
@@ -445,6 +448,51 @@ This  gnerated an HTML5 mobile-friendlyt report, summarising the output of a
...
+```
+
+You can monitor your script in the queue with:
+```
+squeue | grep $USER
+```
+
+You will also see a couple log files appear which will allow you to follow
+the output of the process.
+If you wish to send to to a specific path, you may edit the lineS at the top of
```
Comment (valeriupredoi) [inline README.md:487]:
```suggestion
If you wish to send to to a specific path, you may edit the lines at the top of 
```

File: bgcval2/analysis_compare.py [hunk 3]
```diff
@@ -140,12 +140,52 @@ def apply_shifttimes(mdata, jobID, shifttimes):
...
+    if 0 in [len(times), len(datas), ]:
+       return times, datas
+
+    print('apply_timerange', jobID, timeranges)
+
+    timerange = timeranges.get(jobID, None)
+
+    print('apply_timerange', timerange)    
+    if timerange is None: 
+       print('apply_timerange', timerange, 'is', None)
```
Comment (valeriupredoi) [inline bgcval2/analysis_compare.py:160]:
```suggestion
       print('apply_timerange: timerange is', None) 
```

Comment (valeriupredoi) [reply bgcval2/analysis_compare.py:160]: how it is now would print "apply_timerange None is None" (famous quote by Teresa May)

**Review by valeriupredoi — APPROVED** [review]
cucumber cool stuff man :beer:

**Comment by valeriupredoi** [comment]
all good to merge @ledm :partying_face: You wanna add some of those Terrafirma plots to some documentation? (in a separate PR); also we can get a bgcval2 Github Pages site up if you think that'd be cool (to showcase plots, analyses etc)

**Comment by ledm** [comment]
yes i am!

**Comment by valeriupredoi** [comment]
> yes i am!

yay! Moar nice development :beers:

**Comment by ledm** [comment]
Just you wait for the massive PR I'm about to dump here.

**Comment by valeriupredoi** [comment]
> Just you wait for the massive PR I'm about to dump here.

puppet-awkward

**Pull request merged**